### PR TITLE
Make Resource Editor "Add Existing File..." default path more intuitive

### PR DESCRIPTION
**Pull request by Giftednewt**
This addresses the issue reported in #3492 by defaulting the path to the path of the resource file.

This also caches the last selected path for the resource on a per project basis in the ```VBPackage``` so that it does not reset if you close and re-open the resource editor.

The cache is cleared on solution/project close, project unload and on closing VS.

### CODE REVIEW

File: src/Microsoft.VisualStudio.Editors/package/VBPackage.vb [hunk 1]
```diff
@@ -45,11 +45,22 @@ Namespace Microsoft.VisualStudio.Editors
...
+                Return _stickyProjectResourcePaths
+            End Get
+        End Property
+
         ''' <summary>
         ''' Constructor
         ''' </summary>
         ''' <remarks></remarks>
         Public Sub New()
+            _stickyProjectResourcePaths = New Dictionary(Of Guid, Dictionary(Of String, String))
```
Comment (Pilchie) [inline src/Microsoft.VisualStudio.Editors/package/VBPackage.vb]: Nit: you can just use `As New` in the field definition.

File: src/Microsoft.VisualStudio.Editors/ResourceEditor/ResourceEditorView.vb [hunk 2]
```diff
@@ -4171,24 +4171,37 @@ Namespace Microsoft.VisualStudio.Editors.ResourceEditor
...
             CommitPendingChanges()
 
             Try
                 ' If we can't check out the resource file, do not pop up any dialog...
                 RootDesigner.DesignerLoader.ManualCheckOut()
 
+                Dim resourcePathAndName As String = RootDesigner.GetResXFileNameAndPath()
+                Dim projectGuid As Guid = VBPackage.Instance.ProjectGUID(GetVsHierarchy())
+                Dim addExistingFilePath As String = String.Empty
+                Dim resxDictionary As Dictionary(Of String, String) = Nothing
```
Comment (Pilchie) [inline src/Microsoft.VisualStudio.Editors/ResourceEditor/ResourceEditorView.vb]: Consider renaming to `resxFileToAddExistingBasePath` or something similar.

File: src/Microsoft.VisualStudio.Editors/package/VBPackage.vb [hunk 3]
```diff
@@ -45,11 +45,22 @@ Namespace Microsoft.VisualStudio.Editors
         ' Map between unique project GUID and the last viewed tab in the project designer...
         Private _lastViewedProjectDesignerTab As Dictionary(Of Guid, Byte)
 
+        'Resource File Sticky Paths per project
+        Private _stickyProjectResourcePaths As Dictionary(Of Guid, Dictionary(Of String, String))
+
+        Public ReadOnly Property StickyProjectResourcePaths() As Dictionary(Of Guid, Dictionary(Of String, String))
```
Comment (Pilchie) [inline src/Microsoft.VisualStudio.Editors/package/VBPackage.vb]: Could use an autoprop by leaving out the `Get` etc.

File: src/Microsoft.VisualStudio.Editors/package/VBPackage.vb [hunk 4]
```diff
@@ -45,11 +45,22 @@ Namespace Microsoft.VisualStudio.Editors
         ' Map between unique project GUID and the last viewed tab in the project designer...
         Private _lastViewedProjectDesignerTab As Dictionary(Of Guid, Byte)
 
+        'Resource File Sticky Paths per project
+        Private _stickyProjectResourcePaths As Dictionary(Of Guid, Dictionary(Of String, String))
```
Comment (davkean) [inline src/Microsoft.VisualStudio.Editors/package/VBPackage.vb]: Mark as readonly.

Comment (davkean) [reply src/Microsoft.VisualStudio.Editors/package/VBPackage.vb]: You can just use As New here.

**Comment by davkean** [comment]
This is great - can you mark the field as readonly before we merge? This will be showing up as info message in Error List for 15.7, I literally just cleaned them up.

**Comment by davkean** [comment]
Thanks for the contribution!

**Pull request merged**